addons/mail/models/ir_attachment.py

```diff
@@ -10,6 +10,20 @@
 class IrAttachment(models.Model):
     _inherit = 'ir.attachment'
 
+    @api.model
+    def check(self, mode, values=None):
+        super().check(mode, values=values)
+        if mode not in ('unlink', 'write') or not self or self.env.is_admin():
+            return
+        if self.create_uid == self.env.user:
+            return
```

Comment:
I'm not sure exactly in which scenario we call this method, but we might want to avoid checking the create_uid if it's a public user as they will all share the same user for everyone (when using guest in particular, but not only).

Reply:
Normally public users should only get access to the mail attachments via an access token (?)
So they're effectively sudo when they get to this point and none of this is even called in the first place.

At least I believe that's the case for public chat rooms and portal, maybe I'm missing a use case.
This does raise the issue that the fix doesn't apply to those cases though :x
I'm not sure it's as big of a deal as in a live chat you can just re-send the attachment but I'll check if that can be fixed easily too.